gbpservice/nfp/tests/unit/nfp/configurator/test_firewall.py

```diff
@@ -0,0 +1,887 @@
+        ''' Handle event for configure_routes '''
+        ev = FakeEvent()
+        ev.id = 'CONFIGURE_ROUTES'
+        self._test_handle_event(ev)
+
+    def test_clear_routes_genericconfigeventhandler(self):
+        ''' Handle event for clear_routes '''
+        ev = FakeEvent()
+        ev.id = 'CLEAR_ROUTES'
+        self._test_handle_event(ev)
+
+    def test_configure_hm_initial_genericconfigeventhandler(self):
+        ''' Handle event for configure_hm '''
+        ev = FakeEvent()
+        ev.id = 'CONFIGURE_HEALTHMONITOR initial'
+        self._test_handle_event(ev)
+
+    def test_configure_hm_forever_genericconfigeventhandler(self):
+        ''' Handle event for configure_hm '''
+        ev = FakeEvent()
+        ev.data['kwargs'].update({'periodicity': gen_cfg_const.FOREVER})
+        ev.id = 'CONFIGURE_HEALTHMONITOR forever'
+        self._test_handle_event(ev)
+
+    def test_clear_hm_genericconfigeventhandler(self):
+        ''' Handle event for clear_hm '''
+        ev = FakeEvent()
+        ev.id = 'CLEAR_HEALTHMONITOR'
+        self._test_handle_event(ev)
+
+    def test_handle_configure_healthmonitor_genericconfigeventhandler(self):
+        ''' Handle event for handle_configure_healthmonitor '''
+        ev = FakeEvent()
+        ev.id = 'CONFIGURE_HEALTHMONITOR'
+        self._test_handle_periodic_event(ev)
+
+
+class FwaasHandlerTestCase(unittest.TestCase):
+    ''' Generic Config Handler for Firewall module '''
+
+    def __init__(self, *args, **kwargs):
+        super(FwaasHandlerTestCase, self).__init__(*args, **kwargs)
+        self.fo = FakeObjects()
+        self.ev = FakeEvent()
+
+    @mock.patch(__name__ + '.FakeObjects.nqueue')
+    @mock.patch(__name__ + '.FakeObjects.rpcmgr')
+    @mock.patch(__name__ + '.FakeObjects.drivers')
+    @mock.patch(__name__ + '.FakeObjects.sc')
+    def _get_FwHandler_objects(self, sc, drivers, rpcmgr, nqueue):
+        agent = fw.FWaasEventHandler(sc, drivers, rpcmgr, nqueue)
+        return agent
+
+    def _test_handle_event(self, rule_list_info=True):
+        agent = self._get_FwHandler_objects()
+        driver = FwaasDriver()
+
+        with mock.patch.object(
+             agent.plugin_rpc, 'set_firewall_status') as (
+                                                    mock_set_fw_status), \
+             mock.patch.object(
+                agent.plugin_rpc, 'firewall_deleted') as (mock_fw_deleted), \
+             mock.patch.object(
+                driver, 'create_firewall') as mock_create_fw, \
+             mock.patch.object(
+                driver, 'update_firewall') as mock_update_fw, \
+             mock.patch.object(
+                driver, 'delete_firewall') as mock_delete_fw, \
+             mock.patch.object(
+                agent, '_get_driver', return_value=driver):
+
+            firewall = self.fo._fake_firewall_obj()
+            if not rule_list_info:
+                firewall.update({'firewall_rule_list': False})
```

Comment:
Why is this rule_list added here? Are there two different cases of firewall create here?